bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestBatchExecutor.java

```diff
@@ -197,65 +224,35 @@ public void testGetCallback() throws Exception {
-      @Override
-      public Void call() throws Exception {
-        List<Get> gets = Arrays.asList(new Get(EMPTY_KEY));
-        underTest.batchCallback(gets, new Result[1], callback);
-        return null;
-      }
-    });
+    List<Get> gets = Arrays.asList(new Get(EMPTY_KEY));
+    createExecutor(DEFAULT_OPTIONS).batchCallback(gets, new Result[1], callback);
 
     verify(callback, times(1)).update(same(BatchExecutor.NO_REGION), same(EMPTY_KEY),
       argThat(matchesRow(Adapters.ROW_ADAPTER.adaptResponse(response))));
   }
 
-
-  private Result[] batch(final List<? extends org.apache.hadoop.hbase.client.Row> actions)
-      throws InterruptedException, ExecutionException, TimeoutException {
-    return runBatch(new Callable<Result[]>() {
-      @Override
-      public Result[] call() throws Exception {
-        return underTest.batch(actions);
-      }
-    });
-  }
-
-  private <T> T runBatch(Callable<T> callable)
-      throws InterruptedException, ExecutionException, TimeoutException {
-    ExecutorService executorService = Executors.newSingleThreadExecutor();
-    try {
-      Future<T> callableFuture = executorService.submit(callable);
-      Thread.sleep(100L);
-      for (Runnable runnable : runnables) {
-        runnable.run();
-      }
-      return callableFuture.get(100L, TimeUnit.MILLISECONDS);
-    } finally {
-      executorService.shutdownNow();
+  @Test
+  public void testBatchBulkGets() throws Exception {
+    List<Get> gets = new ArrayList<>(10);
+    List<Row> responses = new ArrayList<>(9);
+    gets.add(new Get(Bytes.toBytes("key0")));
+    for (int i=1; i<10; i++){
+      byte[] row_key = randomBytes(8);
+      gets.add(new Get(Bytes.toBytes("key"+i)));
+      responses.add(Row.newBuilder().setKey(BigtableZeroCopyByteStringUtil.wrap(row_key)).build());
     }
-  }
-
-  private Matcher<Result> matchesRow(final Result expected) {
-    return new BaseMatcher<Result>() {
-
-      @Override
-      public void describeTo(Description description) {
-      }
+    when(mockFuture.get()).thenReturn(responses);
 
-      @Override
-      public boolean matches(Object item) {
-        try {
-          Result.compareResults((Result) item, expected);
-          return true;
-        } catch (Exception e) {
-          return false;
-        }
-      }
+    BatchExecutor underTest = createExecutor(new BigtableOptions.Builder().setUseBulkApi(true).build());
+    final List<? extends org.apache.hadoop.hbase.client.Row> actions = gets;
+    final BatchExecutor executor = underTest;
+    Result[] results =  executor.batch(actions);
+    verify(mockAsyncExecutor, times(1)).readRowsAsync(any(ReadRowsRequest.class));
+    Assert.assertTrue(matchesRow(Result.EMPTY_RESULT).matches(results[0]));
+  }
 
-      @Override
-      public void describeMismatch(Object item, Description mismatchDescription) {
-      }
-    };
+  private Result[] batch(final List<? extends org.apache.hadoop.hbase.client.Row> actions)
```

Comment:
Theres a few places where this method could be used but only some where it is. Use it always instead of 
createExecutor(DEFAULT_OPTIONS).batch ?